pkg/store/bucket_test.go

```diff
@@ -1072,14 +1074,78 @@ func newSeries(t testing.TB, lset labels.Labels, smplChunks [][]sample) storepb.
 	return s
 }
 
-func TestSeries_200kSeriesWithOneSample(t *testing.T) {
-	benchSeries_SeriesWithOneSample(testutil.NewTB(t), 200000)
+func TestSeries(t *testing.T) {
+	tb := testutil.NewTB(t)
+	tb.Run("200e3SeriesWithOneSample", func(tb testutil.TB) {
+		benchSeries(tb, 200e3, seriesDimension, 200e3)
+	})
+	tb.Run("OneSeriesWith200e3Samples", func(tb testutil.TB) {
+		benchSeries(tb, 200e3, samplesDimension, 200e3)
+	})
 }
+
 func BenchmarkSeries(b *testing.B) {
-	benchSeries_SeriesWithOneSample(testutil.NewTB(b), 10000000)
+	tb := testutil.NewTB(b)
+	tb.Run("10e6SeriesWithOneSample", func(tb testutil.TB) {
+		benchSeries(tb, 10e6, seriesDimension, 1, 10, 10e1, 10e2, 10e3, 10e4, 10e5) // This is too big for my machine: 10e6.
+	})
+	tb.Run("OneSeriesWith100e6Samples", func(tb testutil.TB) {
+		// 100e6 samples = ~17361 days with 15s scrape.
+		benchSeries(tb, 100e6, samplesDimension, 1, 10, 10e1, 10e2, 10e3, 10e4, 10e5, 10e6) // This is too big for my machine: 100e6.
+	})
+}
+
+func createBlockWithOneSample(t testutil.TB, dir string, blockIndex int, totalSeries int) (ulid.ULID, []storepb.Series) {
+	fmt.Println("Building block with numSeries:", totalSeries)
+
+	var series []storepb.Series
+	h, err := tsdb.NewHead(nil, nil, nil, 1)
+	testutil.Ok(t, err)
+	defer testutil.Ok(t, h.Close())
+
+	app := h.Appender()
+
+	for i := 0; i < totalSeries; i++ {
+		ts := int64(blockIndex*totalSeries + i)
+		lbls := labels.FromStrings("foo", "bar", "i", fmt.Sprintf("%07d%s", ts, postingsBenchSuffix))
+		series = append(series, newSeries(t, append(labels.Labels{{Name: "ext1", Value: "1"}}, lbls...), [][]sample{{sample{t: ts, v: 0}}}))
+
+		_, err := app.Add(lbls, ts, 0)
+		testutil.Ok(t, err)
+	}
+	testutil.Ok(t, app.Commit())
+
+	return createBlockFromHead(t, dir, h), series
+}
+
+func createBlockWithOneSeries(t testutil.TB, dir string, lbls labels.Labels, blockIndex int, totalSamples int, random *rand.Rand) ulid.ULID {
+	fmt.Println("Building block with one series with numSamples:", totalSamples)
+
+	h, err := tsdb.NewHead(nil, nil, nil, int64(totalSamples))
+	testutil.Ok(t, err)
+	defer testutil.Ok(t, h.Close())
+
+	app := h.Appender()
+
+	ref, err := app.Add(lbls, int64(blockIndex*totalSamples), random.Float64())
+	testutil.Ok(t, err)
+	for i := 1; i < totalSamples; i++ {
+		ts := int64(blockIndex*totalSamples + i)
+		testutil.Ok(t, app.AddFast(ref, ts, random.Float64()))
+	}
+	testutil.Ok(t, app.Commit())
+
+	return createBlockFromHead(t, dir, h)
 }
 
-func benchSeries_SeriesWithOneSample(t testutil.TB, totalSeries int) {
+type Dimension string
```

Comment:
I realize this is a test file but it is still good practice IMO to keep these identifiers lowercase

Reply:
agree

Reply:
Will fix on later PR